### ISSUE TITLE
AzureML pipelines not working anymore with new azure-mgmt-resource release (20.1.0)

### ISSUE THREAD

**Issue by jarandaf** [Azure/azure-sdk-for-python#22785]
- **Package Name**: azureml-sdk
- **Package Version**: 1.31.0
- **Operating System**: Linux
- **Python Version**: 3.8.11

**Describe the bug**

Our Azure ML pipelines started to fail today. There seems to be an issue with  the new `azure-mgmt-resource` package release (20.1.0, released a few hours ago) that was not observed before (until today, our pipelines were resolving this package to version 20.0.0). The following error appears when trying to run a step that uses the Run object:

```python
stderr]Traceback (most recent call last):
[stderr]  File "seas_etl.py", line 7, in <module>
[stderr]    from azureml.core.run import Run
[stderr]  File "/azureml-envs/azureml_fe1cb240319d19bea0cf28e7c9afcfdf/lib/python3.8/site-packages/azureml/core/__init__.py", line 13, in <module>
[stderr]    from .workspace import Workspace
[stderr]  File "/azureml-envs/azureml_fe1cb240319d19bea0cf28e7c9afcfdf/lib/python3.8/site-packages/azureml/core/workspace.py", line 22, in <module>
[stderr]    from azureml._project import _commands
[stderr]  File "/azureml-envs/azureml_fe1cb240319d19bea0cf28e7c9afcfdf/lib/python3.8/site-packages/azureml/_project/_commands.py", line 29, in <module>
[stderr]    from azure.mgmt.resource import ResourceManagementClient
[stderr]  File "/azureml-envs/azureml_fe1cb240319d19bea0cf28e7c9afcfdf/lib/python3.8/site-packages/azure/mgmt/resource/__init__.py", line 9, in <module>
[stderr]    from .managedapplications import ApplicationClient
[stderr]  File "/azureml-envs/azureml_fe1cb240319d19bea0cf28e7c9afcfdf/lib/python3.8/site-packages/azure/mgmt/resource/managedapplications/__init__.py", line 9, in <module>
[stderr]    from ._application_client import ApplicationClient
[stderr]  File "/azureml-envs/azureml_fe1cb240319d19bea0cf28e7c9afcfdf/lib/python3.8/site-packages/azure/mgmt/resource/managedapplications/_application_client.py", line 17, in <module>
[stderr]    from ._configuration import ApplicationClientConfiguration
[stderr]  File "/azureml-envs/azureml_fe1cb240319d19bea0cf28e7c9afcfdf/lib/python3.8/site-packages/azure/mgmt/resource/managedapplications/_configuration.py", line 13, in <module>
[stderr]    from azure.mgmt.core.policies import ARMChallengeAuthenticationPolicy, ARMHttpLoggingPolicy
[stderr]ImportError: cannot import name 'ARMChallengeAuthenticationPolicy' from 'azure.mgmt.core.policies' (/azureml-envs/azureml_fe1cb240319d19bea0cf28e7c9afcfdf/lib/python3.8/site-packages/azure/mgmt/core/policies/__init__.py)
[stderr]
```

**To Reproduce**
1. Create a pipeline with a step that uses the Run object.
2. Run the pipeline
3. Get above error

**Expected behavior**

The pipeline should be able to start the step.

**Comment by tjprescott** [comment]
@msyyc can you take a look?

**Comment by msyyc** [comment]
Hi, please make sure [`azure-mgmt-core` version >= 1.3.0](https://github.com/Azure/azure-sdk-for-python/blob/a5ea4bf7d0decafbe355467ddb0bdd334b9f953b/sdk/resources/azure-mgmt-resource/setup.py#L71) in your environment

**Comment by jarandaf** [comment]
Thank you @msyyc. That solved the issue (we had a pinned version somewhere that avoided `azure-mgmt-core` to be resolved to version >= 1.3).
